dissect/target/tools/query.py

```diff
@@ -110,67 +118,127 @@ def main():
+        for index, task in enumerate(tasklist):
+            collected = {}
+            target, funcs = task
+            for func in funcs:
+                collected[func.name] = func.plugin_desc
+
+            if collected_plugins:
+                common = collected_plugins.keys() & collected.keys()
+                collected_plugins = {plugin_name: collected_plugins[plugin_name] for plugin_name in common}
+            else:
+                collected_plugins = collected
+
+        # Display in a user friendly manner
         t = Target()
         fparser = generate_argparse_for_bound_method(t.plugins, usage_tmpl=USAGE_FORMAT_TMPL)
         fargs, rest = fparser.parse_known_args(rest)
-        t.plugins(**vars(fargs))
-
-        fparser = generate_argparse_for_bound_method(t.loaders, usage_tmpl=USAGE_FORMAT_TMPL)
-        fargs, rest = fparser.parse_known_args(rest)
-        t.loaders(**vars(fargs))
-
+        t.plugins(collected_plugins.values())  # , True, **vars(fargs)
+        # No real targets specified, user might be interested to see available loaders
+        if len(tasklist) == 1 and not tasklist[0][0]._applied:
+            fparser = generate_argparse_for_bound_method(t.loaders, usage_tmpl=USAGE_FORMAT_TMPL)
+            fargs, rest = fparser.parse_known_args(rest)
+            t.loaders(**vars(fargs))
         parser.exit()
 
-    if not args.function:
-        parser.error("argument -f/--function is required")
-
-    functions = args.function.split(",")
+    # If we get here, that means we need to execute the collected plugins,
+    # first check if the output types are uniform, if not:
+    # - emit warning
+    # - fall back to record and
+    # - ignore further errors of this kind
+    total_num_tasks = 0
+    output_types = set()
+
+    for task in tasklist:
+        target, funcs = task
+        for func in funcs:
+            total_num_tasks += 1
+            output_types.add(func.output_type)
+    # Important! By setting the default output to record we also
+    # indicate that further errors because of mixed output types will
+    # be supressed (makes no sense anymore, we already have set the default)
+    default_output_type = None
+    if len(output_types) > 1:
+        log.warning(f"Mixed output types detected: {'/'.join(output_types)}, only outputting records.")
+        default_output_type = "record"
 
     # Show help for a function
     if "-h" in rest or "--help" in rest:
-        first_func = functions[0]
-        obj = get_attr_for_attr_path(first_func)
-
+        if not total_num_tasks:
+            parser.error("Function(s) not found, see -l for available plugins")
+            parser.exit()
+        target, func_list = tasklist[0]
+        func = func_list[0]
+        obj = getattr(func.class_object, func.method_name)
         if not obj:
-            parser.error(f"Can't find plugin with function `{first_func}`")
-
+            parser.error(f"Can't find plugin with function `{func.method_name}`")
```

Comment:
Is this even possible wuth `tasklist` being build up using `target.find_plugin_functions()`?

Reply:
Should not be, but otherwise the error message will probably be quite cryptic...

Reply:
Due to a programming error the value of obj can still be Falsy (like None)?

Reply:
Done